homeassistant/components/ezviz/camera.py

```diff
@@ -0,0 +1,236 @@
+        super().__init__()
+
+        self._username = data["username"]
+        self._password = data["password"]
+        self._rtsp_stream = data["rtsp_stream"]
+
+        self._ezviz_camera = data["ezviz_camera"]
+        self._serial = data["serial"]
+        self._name = data["name"]
+        self._status = data["status"]
+        self._privacy = data["privacy"]
+        self._audio = data["audio"]
+        self._ir_led = data["ir_led"]
+        self._state_led = data["state_led"]
+        self._follow_move = data["follow_move"]
+        self._alarm_notify = data["alarm_notify"]
+        self._alarm_sound_mod = data["alarm_sound_mod"]
+        self._encrypted = data["encrypted"]
+        self._local_ip = data["local_ip"]
+        self._detection_sensibility = data["detection_sensibility"]
+        self._device_sub_category = data["device_sub_category"]
+        self._local_rtsp_port = data["local_rtsp_port"]
+
+        self._ffmpeg = None
+
+    def update(self):
+        """Update the camera states."""
+
+        data = self._ezviz_camera.status()
+
+        self._name = data["name"]
+        self._status = data["status"]
+        self._privacy = data["privacy"]
+        self._audio = data["audio"]
+        self._ir_led = data["ir_led"]
+        self._state_led = data["state_led"]
+        self._follow_move = data["follow_move"]
+        self._alarm_notify = data["alarm_notify"]
+        self._alarm_sound_mod = data["alarm_sound_mod"]
+        self._encrypted = data["encrypted"]
+        self._local_ip = data["local_ip"]
+        self._detection_sensibility = data["detection_sensibility"]
+        self._device_sub_category = data["device_sub_category"]
+        self._local_rtsp_port = data["local_rtsp_port"]
+
+    async def async_added_to_hass(self):
+        """Subscribe to ffmpeg and add camera to list."""
+        self._ffmpeg = self.hass.data[DATA_FFMPEG]
+        entities = self.hass.data.setdefault(EZVIZ_DATA, {}).setdefault(ENTITIES, [])
+        entities.append(self)
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return True
+
+    @property
+    def device_state_attributes(self):
+        """Return the Ezviz-specific camera state attributes."""
+        return {
+            "serial": self._serial,
+            "name": self._name,
+            "status": self._status,
+            "device_sub_category": self._device_sub_category,
+            "privacy": self._privacy,
+            "audio": self._audio,
+            "ir_led": self._ir_led,
+            "state_led": self._state_led,
+            "follow_move": self._follow_move,
+            "alarm_notify": self._alarm_notify,
+            "alarm_sound_mod": self._alarm_sound_mod,
+            "encrypted": self._encrypted,
+            "local_ip": self._local_ip,
+            "detection_sensibility": self._detection_sensibility,
```

Comment:
Please describe all these state attributes and especially in regards to their dynamic nature or lack thereof. There needs to be a motivation to add each one.

Reply:
# device is connected & working
            "status": self._status,
            # if privacy == true, the device closed the lid or did a 180° tilt
            "privacy": self._privacy,
            # is the camera listening ?
            "audio": self._audio,
            # infrared led on ?
            "ir_led": self._ir_led,
            # state led on  ?
            "state_led": self._state_led,
            # if true, the camera will move automatically to follow movements
            "follow_move": self._follow_move,
            # if true, if some movement is detected, the app is notified
            "alarm_notify": self._alarm_notify,
            # if true, if some movement is detected, the camera makes some sound
            "alarm_sound_mod": self._alarm_sound_mod,
            # are the camera's stored videos/images encrypted?
            "encrypted": self._encrypted,
            # camera's local ip on local network
            "local_ip": self._local_ip,
            # from 1 to 9, the higher is the sensibility, the more it will detect small movemements
            "detection_sensibility": self._detection_sensibility,

Reply:
I think most seem good. Will the local ip ever change during entity lifetime? Can we remove that? Is it needed for automations?

Reply:
indeed it can change, if for instance the DHCP of the user is not providing a static IP to the camera